File: Level 0/코딩테스트 입문/다항식 더하기/solution.py
def solution(polynomial):
    x_num = 0
    num = 0

    for n in polynomial.split(" + "):
        if "x" in n:
            x_num += int(n.replace("x", "") or 1)
        else:
            num += int(n)

    answer = []

    if x_num:
        answer.append(f"{x_num if x_num != 1 else ''}x")

    if num:
        answer.append(str(num))

    return " + ".join(answer)

File: Level 0/코딩테스트 입문/다항식 더하기/solution.js
function solution(polynomial) {
  let xNum = 0;
  let num = 0;

  polynomial.split(" + ").forEach((n) => {
    if (n.includes("x")) {
      xNum += +n.replaceAll("x", "") || 1;
    } else {
      num += +n;
    }
  });

  const answer = [];

  if (xNum) answer.push(`${xNum === 1 ? "" : xNum}x`);
  if (num) answer.push(num);

  return answer.join(" + ");
}
